src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -66,6 +66,25 @@ std::string GenKeyRedisMatchPattern(const std::string &external_storage_namespac
                       "*");
 }
 
+std::vector<std::vector<std::string>> GenCommandsBatched(
+    const std::string &command,
+    const std::string &hash_field,
+    const std::vector<std::string> &keys) {
+  std::vector<std::vector<std::string>> batched_requests;
+  for (auto &key : keys) {
+    // If it's empty or the last batch is full, add a new batch.
+    if (batched_requests.empty() ||
+        batched_requests.back().size() - 1 ==
```

Comment:
will this ever underflow?

Reply:
I think it shouldn't, but I agree this part is bad. Let me update it.

Reply:
But let me update it to avoid underflow.